cli/azd/cmd/auth_login.go

```diff
@@ -462,48 +464,47 @@ func (la *loginAction) login(ctx context.Context) error {
 		//
 		// TODO(ellismg): We may want instead to call some explicit `/login` endpoint on the external auth system instead
 		// of abusing the token request in this manner. This would allow the other end to provide a more tailored experience.
-		if _, err := la.verifyLoggedIn(ctx); err != nil {
-			return err
-		}
-	} else {
-		useDevCode, err := parseUseDeviceCode(ctx, la.flags.useDeviceCode, la.commandRunner)
-		if err != nil {
-			return err
-		}
+		_, err := la.verifyLoggedIn(ctx)
+		return err
+	}
 
-		if useDevCode {
-			_, err := la.authManager.LoginWithDeviceCode(ctx, la.flags.tenantID, la.flags.scopes, func(url string) error {
-				if !la.flags.global.NoPrompt {
-					la.console.Message(ctx, "Then press enter and continue to log in from your browser...")
-					la.console.WaitForEnter()
-					openWithDefaultBrowser(ctx, la.console, url)
-					return nil
-				}
-				// For no-prompt, Just provide instructions without trying to open the browser
-				// If manual browsing is enabled, we don't want to open the browser automatically
-				la.console.Message(ctx, fmt.Sprintf("Then, go to: %s", url))
+	useDevCode, err := parseUseDeviceCode(ctx, la.flags.useDeviceCode, la.commandRunner)
+	if err != nil {
+		return err
+	}
+	if useDevCode {
+		_, err = la.authManager.LoginWithDeviceCode(ctx, la.flags.tenantID, la.flags.scopes, func(url string) error {
+			if !la.flags.global.NoPrompt {
+				la.console.Message(ctx, "Then press enter and continue to log in from your browser...")
+				la.console.WaitForEnter()
+				openWithDefaultBrowser(ctx, la.console, url)
 				return nil
-			})
-			if err != nil {
-				return fmt.Errorf("logging in: %w", err)
 			}
-		} else {
-			_, err := la.authManager.LoginInteractive(ctx, la.flags.scopes,
-				&auth.LoginInteractiveOptions{
-					TenantID:     la.flags.tenantID,
-					RedirectPort: la.flags.redirectPort,
-					WithOpenUrl: func(url string) error {
-						openWithDefaultBrowser(ctx, la.console, url)
-						return nil
-					},
-				})
-			if err != nil {
-				return fmt.Errorf("logging in: %w", err)
-			}
-		}
+			// For no-prompt, Just provide instructions without trying to open the browser
+			// If manual browsing is enabled, we don't want to open the browser automatically
+			la.console.Message(ctx, fmt.Sprintf("Then, go to: %s", url))
+			return nil
+		})
+		return err
 	}
 
-	return nil
+	if oneauth.Supported {
+		err = la.authManager.LoginWithOneAuth(ctx, la.flags.scopes, la.flags.global)
```

Comment:
As you might recall, @chlowell, we added `WithOpenUrl` to the MSAL library so that consumers could control launching the browser.  We needed this because our idea of a default browser differed from the MSAL library, and in some cases (like codespaces) we needed more control over what browser was going to be launched.

Does OneAuth provide a way to control what browser is used for an interactive login?  Or are they hosting thier own control to handle the login flow instead of delegating to the browser?

Reply:
It's the latter--OneAuth creates a new window instead of opening a browser.